docs/guides/server/db.adoc

```diff
@@ -128,20 +128,51 @@ Then continue configuring the database as described in the next section.
 
 For each supported database, the server provides some opinionated defaults to simplify database configuration. You complete the configuration by providing some key settings such as the database host and credentials.
 
-. Start the server and set the basic options to configure a database
+The configuration can be set during a *build command* OR a *start command*:
+
+. Using a *build command* followed by an optimzed start command (recommended)
++
+First, the minimum settings needed to connect to the database can be specified in *conf/keycloak.conf*:
++
+----
+# The database vendor.
+db=postgres
+
+# The username of the database user.
+db-username=keycloak
+
+# The password of the database user.
+db-password=change_me
+
+# Sets the hostname of the default JDBC URL of the chosen vendor
+db-url-host=keycloak-postgres
+----
++
+
+Then, the following commands create a new and optimized server image based on the configuration options and start the server.
 +
-<@kc.start parameters="--db postgres --db-url-host mypostgres --db-username myuser --db-password change_me"/>
+----
+bin/kc.[sh|bat] build
+bin/kc.[sh|bat] start --optimized
+----
 +
 
-This command includes the minimum settings needed to connect to the database.
+. Using *only a start command* (without --optimized)
++
+<@kc.start parameters="--db postgres --db-url-host keycloak-postgres --db-username keycloak --db-password change_me"/>
++
+
+This command includes the minimum settings needed to connect to the database but it exposes the database password and is not recommended. Use the `conf/keycloak.conf` as shown above, environment variables, or keystore for at least the password.
 
 The default schema is `keycloak`, but you can change it by using the `db-schema` configuration option.
 
-[WARNING]
-Do NOT use the `--optimized` flag for the `start` command if you want to use a particular DB (except the H2).
-Executing the build phase before starting the server instance is necessary.
-You can achieve it either by starting the instance without the `--optimized` flag,
-or by executing the `build` command before the optimized start.
+It is also possible to configure the database when https://www.keycloak.org/server/importExport[importing or exporting Realms] or https://www.keycloak.org/server/bootstrap-admin-recovery[creating new Keycloak admin users / service accounts] (starting with release 26):
```

Comment:
```suggestion
It is also possible to configure the database when <@links.server id="importExport"/> or <@links.server id="bootstrap-admin-recovery"/>:
```

Reply:
without mentioning '(starting with release 26)'?

Reply:
> without mentioning '(starting with release 26)'?

Yes, we don't typically do that as the docs are versioned as well.